# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/loader/ModuleLoadRequest.h
# Commit: c6c863899bf9
# Full Hash: c6c863899bf944a2cb49531cd7a9a7c752b6f700
# Author: Yoshi Cheng-Hao Huang <allstars.chh@gmail.com>
# Date: 2025-09-26 06:07 +0000
# Regressor Bug: 1968890
# File Overlap Count: 3
# Overlapping Files: js/loader/ModuleLoadRequest.h, js/loader/ModuleLoadRequest.cpp, js/loader/ModuleLoaderBase.cpp
# Description:
#   Bug 1968890 - Part 5: Add ModuleLoadRequest::SetErroredLoadingImports(). r=jonco
# ==============================================================================

diff -r 52a87b60e974 -r c6c863899bf9 js/loader/ModuleLoadRequest.h
--- a/js/loader/ModuleLoadRequest.h	Fri Sep 26 06:07:44 2025 +0000
+++ b/js/loader/ModuleLoadRequest.h	Fri Sep 26 06:07:44 2025 +0000
@@ -118,9 +118,19 @@
 
   const Kind mKind;
 
+  void SetErroredLoadingImports() {
+    MOZ_ASSERT(IsDynamicImport());
+    MOZ_ASSERT(IsFetching());
+    mErroredLoadingImports = true;
+  }
+
   // Type of module (JavaScript, JSON)
   const ModuleType mModuleType;
 
+  // A flag (for dynamic import) that indicates the module script is
+  // successfully fetched and compiled, but its dependencies are failed to load.
+  bool mErroredLoadingImports;
+
   // Pointer to the script loader, used to trigger actions when the module load
   // finishes.
   RefPtr<ModuleLoaderBase> mLoader;